### DISCUSSION TITLE
How to hide a field on update in simple view

### DISCUSSION THREAD

**Discussion by bomshteyn** [filamentphp/filament#3171] [Help]
When using a simple view (modals only), how can `hiddenOn` & `visibleOn` be used to hide a field on update but show on create?

the documentation gives an example of hiding on password:

```PHP
Forms\Components\TextInput::make('password')
    ->password()
    ->required()
    ->hiddenOn(Pages\EditUser::class),
```

But since here the field is shown on same page there is no separate page.

**Comment by danharrin** [comment]
`->hidden(fn (ListRecords $livewire) => $livewire->mountedTableAction === 'edit')`

**Reply by bomshteyn** [reply]
Thank you, one issue with this.
This works if you click the new button, and then click edit. But if i click the edit button then cancel and click new the field is hidden.

**Reply by bomshteyn** [reply]
Also interesting thing i noticed while trying to debug. The hidden method is being triggered 3 times when i click the edit button in the simple view.

**Reply by danharrin** [reply]
The method being triggered multiple times is not a problem, there are just several checks at different stages in the lifecycle

> But if i click the edit button then cancel and click new the field is hidden.

Not sure what you mean, do you have a screen recording

**Reply by bomshteyn** [reply]
Here is a screen recording: https://www.loom.com/share/00383255bb3747ce95f80d7837fd33ef

**Reply by bomshteyn** [reply]
Basically once you hit the edit button, the  `$livewire->mountedTableAction` is set to `'edit'` and it doesn't get unset when clicking cancel / closing modal.

**Reply by danharrin** [reply]
I see, yeah. Thats weird that it doesnt get reset to `create`, but staying as `edit` when it closes is inconsequential tbh

**Reply by bomshteyn** [reply]
@danharrin The issue seems to stem from the fact that the create button is a page action, not table action. So when clicking the create button it's updating `mountedAction` not `mountedTableAction`

**Reply by bomshteyn** [reply]
I have created a pull request to address this #3345
